### PR TITLE
touchdetector: update development dependencies.

### PR DESCRIPTION
**Pull request by matz-e**

### CODE REVIEW

File: var/spack/repos/builtin/packages/touchdetector/package.py [hunk 1]
```diff
@@ -40,15 +40,17 @@ class Touchdetector(CMakePackage):
...
-    depends_on('hpctools~openmp', when='~openmp')
-    depends_on('hpctools+openmp', when='+openmp')
     depends_on('morphio@2.0.8:', when='@4.5:')
     depends_on('mvdtool@1.5.1:', when='@4.5:')
     depends_on('mpi')
     depends_on('pugixml', when='@4.5:')
 
     # Old dependencies
+    depends_on('hpctools~openmp', when='~openmp@:4.4')
+    depends_on('hpctools+openmp', when='+openmp@:4.4')
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/touchdetector/package.py:53]: I thought this was submodule.

File: var/spack/repos/builtin/packages/touchdetector/package.py [hunk 2]
```diff
@@ -40,15 +40,17 @@ class Touchdetector(CMakePackage):
 
     depends_on('cmake', type='build')
     depends_on('boost@1.50:')
+    depends_on('eigen', when='@4.5:')
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/touchdetector/package.py:43]: eigen has following `True` variants: `+fftw+metis+mpfr+scotch+suitesparse`. Most likely these are not needed for touch detector (same as NMODL). could you turn them off here?

**Review by pramodk — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/touchdetector/package.py (hunk 1 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/touchdetector/package.py:53]: No.

File: var/spack/repos/builtin/packages/touchdetector/package.py (hunk 2 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/touchdetector/package.py:43]: OK.

Comment (matz-e) [reply var/spack/repos/builtin/packages/touchdetector/package.py:43]: > eigen has following `True` variants: `+fftw+metis+mpfr+scotch+suitesparse`. Most likely these are not needed for touch detector (same as NMODL). could you turn them off here?

Do you have this recipe somewhere private?

**Review by pramodk — APPROVED** [review]

**Pull request merged**